### PR TITLE
Issue #905: Switch codebase to Monorepo

### PR DESCRIPTION
**Pull request by phaumer**
## Proposed changes

Details and discussion in <https://github.com/zowe/vscode-extension-for-zowe/issues/905>

## Release Notes

- Converted source code to monorepo using yarn and extracted extender's API to importable npm package.

Milestone:

Changelog:

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [ ] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [ ] `npm run vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zdmullen** [comment]
To be demoed Tuesday, August 25

**Comment by VitGottwald** [comment]
Due to several of commits that were left unsigned, I had to squash the this branch into a single commit and force push it. My apologies @phaumer, @JillieBeanSim, and others.

The commit before the squash was **aa6615f** as the PR history shows. The current commit is **17c0233**. Anyone can verify that the contents of the files has not changed by running `git diff aa6615f 17c0233`.

**Comment by phaumer** [comment]
@VitGottwald I don't think a force push into someone else's branch is a good practice. If you are signing with a gmail address what is the point anyway? You could have added something to the dco_signoffs folder.

**Comment by VitGottwald** [comment]
Hi Peter, sorry for that. What is dco_signoffs folder?

**Comment by VitGottwald** [comment]
After a discussion with Peter, I restored the branch (see the force push above). And I am going to add a record to the dco_signoffs folder.

File: packages/zowe-explorer-api/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  "repository": {
+    "url": "https://github.com/zowe/vscode-extension-for-zowe"
+  },
+  "main": "lib/index.js",
+  "files": [
+    "lib",
+    "src"
+  ],
+  "scripts": {
+    "build": "yarn clean && yarn license && tsc -p ./",
```
Comment (dkelosky) [inline packages/zowe-explorer-api/package.json:17]: should `yarn` be in `devDependencies` so this runs a common version of `yarn` in `node_modules`?

**Comment by jellypuno** [comment]
Here arethe issues that I've found:
- Profile used: z/OSMF Profile with SCS

This error message appeared during initialization:
image

Names for Allocate Like and Validation is not found:
image

image

image

If Global Validation is true, filter prompt does not work:
monoFilter

**Comment by zFernand0** [comment]
9679017d7e34b3de89c3d18b031000491ce89060 should resolve missing messages/descriptions https://github.com/zowe/vscode-extension-for-zowe/pull/981#issuecomment-714475684

**Review by zFernand0 — DISMISSED** [review]
LGTM!

**Comment by zFernand0** [comment]
Unit tests are passing, but the workflow fails to save the results.
We can address this in a subsequent PR 😋

**Review by zFernand0 — APPROVED** [review]

**Pull request merged**